src/overlays/actors/ovl_Bg_Jya_Lift/z_bg_jya_lift.c

```diff
@@ -26,24 +34,128 @@ const ActorInit Bg_Jya_Lift_InitVars = {
+    D_8089A020 = 1;
+    this->unk_16A = 1;
+}
+
+void BgJyaLift_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaLift* this = THIS;
+
+    if (this->unk_16A != 0) {
+
+        // Goddess Lift DT
+        osSyncPrintf("女神リフト DT\n");
+        D_8089A020 = 0;
+        DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    }
+}
+
+void BgJyaLift_SetInitPosY(BgJyaLift* this) {
+    this->actionFunc = BgJyaLift_DelayMove;
+    this->dyna.actor.posRot.pos.y = 1613.0f;
+    this->moveDelay = 0;
+}
+
+void BgJyaLift_DelayMove(BgJyaLift* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F) || (this->moveDelay > 0)) {
+        this->moveDelay++;
+        if (this->moveDelay >= 20) { 
+            func_800800F8(globalCtx, 0xD66, -0x63, &this->dyna.actor, 0);
+            BgJyaLift_SetupMove(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/BgJyaLift_Destroy.s")
+void BgJyaLift_SetupMove(BgJyaLift* this) {
+    this->actionFunc = BgJyaLift_Move;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899D18.s")
+void BgJyaLift_Move(BgJyaLift* this, GlobalContext* globalCtx) {
+    f32 distFromBottom;
+    f32 tempVelocity;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899D38.s")
+    Math_SmoothScaleMaxMinF(&this->dyna.actor.velocity.y, 4.0f, 0.1f, 1.0f, 0.0f);
+    tempVelocity = (this->dyna.actor.velocity.y < 0.2f) ? 0.2f : this->dyna.actor.velocity.y;
+    distFromBottom = Math_SmoothScaleMaxMinF(&this->dyna.actor.posRot.pos.y, 973.0f, 0.1f, tempVelocity, 0.2f);
+    if ((this->dyna.actor.posRot.pos.y < 1440.0f) && (1440.0f <= this->dyna.actor.pos4.y)) {
+        func_8005B1A4(ACTIVE_CAM);
+    }
+    if (fabsf(distFromBottom) < 0.001f) {
+        BgJyaLift_SetFinalPosY(this);
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_ELEVATOR_STOP);
+    } else {
+        func_8002F974(&this->dyna.actor, 0x200C);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899DBC.s")
+void BgJyaLift_SetFinalPosY(BgJyaLift* this) {
+    this->actionFunc = NULL;
+    this->dyna.actor.posRot.pos.y = 973.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899DCC.s")
+void BgJyaLift_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaLift* this = THIS;
+    GlobalContext* globalCtx2 = globalCtx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899EF0.s")
+    if (this->actionFunc) {
+        this->actionFunc(this);
+    }
+    if ((this->dyna.unk_160 & 4) && ((this->unk_16B & 4) == 0)) {
+        func_8005A77C(globalCtx2->cameraPtrs[0], 0x3F);
+    } else {
+        if (((this->dyna.unk_160) & 4) == 0 && ((this->unk_16B & 4)) && (globalCtx2->cameraPtrs[0]->setting == 0x3F)) {
+            func_8005A77C(globalCtx2->cameraPtrs[0], 3);
+        }
+    }
```

Comment:
I suggest using an else if here and cleaning things up like this:
```suggestion
    if ((this->dyna.unk_160 & 4) && !(this->unk_16B & 4)) {
        func_8005A77C(globalCtx2->cameraPtrs[0], 0x3F);
    } else if (!(this->dyna.unk_160 & 4) && (this->unk_16B & 4) && (globalCtx2->cameraPtrs[0]->setting == 0x3F)) {
        func_8005A77C(globalCtx2->cameraPtrs[0], 3);
    }
```